src/document_add.c

```diff
@@ -343,8 +343,9 @@ static int doAddHashCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int a
   if (rv == AC_OK) {
     // OK. No error
   } else if (rv == AC_ERR_ENOENT) {
-    QueryError_SetErrorFmt(&status, QUERY_EADDARGS, "Unknown keyword: `%s`",
-                           AC_GetStringNC(&ac, NULL));
+    const char *keyword = AC_GetStringNC(&ac, NULL);
+    QueryError_SetErrorFmt(&status, QUERY_EADDARGS, "Unknown keyword: `%s`", keyword);
+    free(keyword);
```

Comment:
AC_GetStringNC does not allocating any new data, why it need to be free? also if you free it with free it means that something was allocated with malloc and not rm_malloc which is bad ..

Reply:
outdated... it was WIP